mmv1/third_party/terraform/resources/resource_apigee_flowhook.go

```diff
@@ -0,0 +1,239 @@
+package google
+
+import (
+	"fmt"
+	"log"
+	"reflect"
+	"time"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func ResourceApigeeFlowhook() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceApigeeFlowhookCreate,
+		Read:   resourceApigeeFlowhookRead,
+		Delete: resourceApigeeFlowhookDelete,
+
+		Importer: &schema.ResourceImporter{
+			State: resourceApigeeFlowhookImport,
+		},
+
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(20 * time.Minute),
+			Delete: schema.DefaultTimeout(20 * time.Minute),
+		},
+
+		Schema: map[string]*schema.Schema{
+			"description": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: `Description of the flow hook.`,
+			},
+			"environment": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: `The resource ID of the environment.`,
+			},
+			"flow_hook_point": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: `Where in the API call flow the flow hook is invoked. Must be one of PreProxyFlowHook, PostProxyFlowHook, PreTargetFlowHook, or PostTargetFlowHook.`,
+			},
+			"org_id": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: `The Apigee Organization associated with the environment`,
+			},
+			"sharedflow": {
+				Type:        schema.TypeString,
+				Required:    true,
```

Comment:
I believe this should be `Optional: true` as the documentation allows an empty string to represent none.